app/components/language-select.js

```diff
@@ -3,30 +3,40 @@ import Ember from 'ember';
 export default Ember.Component.extend({
   config: Ember.inject.service(),
   i18n: Ember.inject.service(),
+  languagePreference: Ember.inject.service(),
 
-  languageOptions: function() {
+  classNames: ['language-select'],
+
+  languageOptions: Ember.computed('i18n.locale', function() {
     let i18n = this.get('i18n');
     return i18n.get('locales').map((item) => {
       return {
         id: item,
         name: i18n.t(`languages.${item}`)
       };
     });
-  }.property('currentLanguage'),
+  }),
 
   onFinish: null,
 
-  _setUserLanguage(language) {
+  _storeUserI18n(i18n) {
     let configDB = this.get('config.configDB');
+    let username;
     configDB.get('current_user').then((user) => {
-      user.i18n = language;
-      configDB.put(user);
+      username = (typeof user.value === 'string') ? user.value : user.value.name;
+      configDB.get('preferences').then((db) => {
```

Comment:
this line should be:
```js
configDB.get('preferences').then((preferences) => {
``` 
As it is getting a reference to a record, not the whole db.